File: A3/A3-022/A3-022.py
# Input light interval and mark as lighted
N = int(input())

degree = [False] * 360
for _ in range(N):
    A, B = map(int, input().split())
    if(A > B):
        B += 360
    
    for i in range(A, B):
        if(not degree[i % 360]):
            degree[i % 360] = True

# Find maximum lighted interval
# Case 1: Maximum interval is [A, B] when A < B
degree_count = 0
max_degree_count = 0
for i in range(360):
    if(degree[i]):
        degree_count += 1
    else:
        max_degree_count = max(max_degree_count, degree_count)
        degree_count = 0
max_degree_count = max(max_degree_count, degree_count)

# Case 2: Maximum interval is [A, B] when A > B
if(max_degree_count < 360):
    A = 359
    while(degree[A]):
        A -= 1
    B = 0
    while(degree[B]):
        B += 1
    degree_count = (B + 360) - A - 1
max_degree_count = max(max_degree_count, degree_count)

# Output
print(max_degree_count)

File: A3/A3-022/A3-022.cpp
#include<iostream>
using namespace std;

bool degree[360];
int N, degCount = 0, maxDegCount = 0;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input light interval and mark as lighted
    cin >> N;
    while(N--)
    {
        int A, B;
        cin >> A >> B;
        if(A > B) { B += 360; }
        
        for(int i = A; i < B; i++)
        {
            if(!degree[i % 360]) { degree[i % 360] = true; }
        }
    }

    // Find maximum lighted interval
    // Case 1: Maximum interval is [A, B] when A < B
    for(int i = 0; i < 360; i++)
    {
        if(degree[i]) { degCount++; }
        else
        { 
            maxDegCount = max(maxDegCount, degCount);
            degCount = 0;
        }
    }
    maxDegCount = max(maxDegCount, degCount);

    // Case 2: Maximum interval is [A, B] when A > B
    if(maxDegCount < 360)
    {
        int A = 359, B = 0;
        while(degree[A]) { A--; }
        while(degree[B]) { B++; }
        degCount = (B + 360) - A - 1;
    }
    maxDegCount = max(maxDegCount, degCount);

    // Output
    cout << maxDegCount << '\n';
    return 0;
}